operator-framework-core/src/main/java/io/javaoperatorsdk/operator/LeaderElectionManager.java

```diff
@@ -31,24 +38,42 @@ public LeaderElectionManager(ControllerManager controllerManager) {
 
   public void init(LeaderElectionConfiguration config, KubernetesClient client) {
     this.identity = identity(config);
-    Lock lock = new LeaseLock(config.getLeaseNamespace(), config.getLeaseName(), identity);
+    final var leaseNamespace =
+        config.getLeaseNamespace().or(LeaderElectionManager::tryGetInClusterNamespace);
+    if (leaseNamespace.isEmpty()) {
+      final var message =
+          "Lease namespace is not set and cannot be inferred. Leader election cannot continue.";
+      log.error(message);
+      throw new IllegalArgumentException(message);
+    }
+    final var lock = new LeaseLock(leaseNamespace.orElseThrow(), config.getLeaseName(), identity);
```

Comment:
Not sure which code style this repo prefers, so this change just tries to align with the majority cases.